core/src/layers/blocking.rs

```diff
@@ -292,6 +292,21 @@ impl<I: oio::Write + 'static> oio::BlockingWrite for BlockingWrapper<I> {
     }
 }
 
+impl<I: oio::Write + 'static> Drop for BlockingWrapper<I> {
+    fn drop(&mut self) {
+        // Attempt to close the inner writer, blocking on the current handle.
+        // This might panic if called from within an existing `block_on` scope
+        // on the same runtime handle.
+        // We ignore the result because drop should not panic.
+        if let Err(err) = self.handle.block_on(self.inner.close()) {
```

Comment:
Calling `close` in `drop` is incorrect. Instead, we should take the `inner` value and drop it within an async context.

To implement this, maybe we use change `inner: I` to `inner: Option<I>` instead.

Reply:
> does this require a `Send` on `I` ?

I'm guessing it's not.